daemon/daemon.go

```diff
@@ -494,7 +494,7 @@ func (daemon *Daemon) restore(cfg *configStore) error {
 				mapLock.Lock()
 				restartContainers[c] = make(chan struct{})
 				mapLock.Unlock()
-			} else if c.HostConfig != nil && c.HostConfig.AutoRemove {
+			} else if c.HostConfig != nil && c.HostConfig.AutoRemove && !cfg.LiveRestoreEnabled {
```

Comment:
The original commit had a check if the container was still running (or paused); https://github.com/moby/moby/commit/3c2886d8a45d8e79b00ab413d91f1af871b58d0a#diff-6a5ec84e73a526676a7b1e1f5e730211f009679a8736b7e7cda6a111a986ab4cR203-R205

```go
if !c.IsRunning() && !c.IsPaused() {
	if daemon.configStore.AutoRestart && c.ShouldRestart() {
		mapLock.Lock()
		restartContainers[c] = make(chan struct{})
		mapLock.Unlock()
	} else if c.HostConfig != nil && c.HostConfig.AutoRemove {
		removeContainers[c.ID] = c
	}
}
```


I wonder if that's the bit that got lost, because there's multiple loops / checks (so perhaps we now fall through to code that shouldn't be run on "running" containers

Reply:
Yes, we still should remove the container which exited while the engine was down.
Added the check and an integration test for this case.